File: software/examples/perl/example_simple.pl
#!/usr/bin/perl

use strict;
use Tinkerforge::IPConnection;
use Tinkerforge::BrickletEnergyMonitor;

use constant HOST => 'localhost';
use constant PORT => 4223;
use constant UID => 'XYZ'; # Change XYZ to the UID of your Energy Monitor Bricklet

my $ipcon = Tinkerforge::IPConnection->new(); # Create IP connection
my $em = Tinkerforge::BrickletEnergyMonitor->new(&UID, $ipcon); # Create device object

$ipcon->connect(&HOST, &PORT); # Connect to brickd
# Don't use device before ipcon is connected

# Get current energy data
my ($voltage, $current, $energy, $real_power, $apparent_power, $reactive_power,
    $power_factor, $frequency) = $em->get_energy_data();

print "Voltage: " . $voltage/100.0 . " V\n";
print "Current: " . $current/100.0 . " A\n";
print "Energy: " . $energy/100.0 . " Wh\n";
print "Real Power: " . $real_power/100.0 . " h\n";
print "Apparent Power: " . $apparent_power/100.0 . " VA\n";
print "Reactive Power: " . $reactive_power/100.0 . " var\n";
print "Power Factor: " . $power_factor/1000.0 . "\n";
print "Frequency: " . $frequency/100.0 . " Hz\n";

print "Press key to exit\n";
<STDIN>;
$ipcon->disconnect();

File: software/examples/ruby/example_simple.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_energy_monitor'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your Energy Monitor Bricklet

ipcon = IPConnection.new # Create IP connection
em = BrickletEnergyMonitor.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Get current energy data as [voltage, current, energy, real_power, apparent_power,
#                             reactive_power, power_factor, frequency]
energy_data = em.get_energy_data

puts "Voltage: #{energy_data[0]/100.0} V"
puts "Current: #{energy_data[1]/100.0} A"
puts "Energy: #{energy_data[2]/100.0} Wh"
puts "Real Power: #{energy_data[3]/100.0} h"
puts "Apparent Power: #{energy_data[4]/100.0} VA"
puts "Reactive Power: #{energy_data[5]/100.0} var"
puts "Power Factor: #{energy_data[6]/1000.0}"
puts "Frequency: #{energy_data[7]/100.0} Hz"

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect
